source/_addonStore/dataManager.py

```diff
@@ -195,23 +219,23 @@ def getLatestAddons(
 			self,
 			onDisplayableError: Optional[DisplayableError.OnDisplayableErrorT] = None,
 	) -> "AddonGUICollectionT":
+		cacheHash = self._getCacheHash()
 		shouldRefreshData = (
 			not self._latestAddonCache
-			or _DataManager._cachePeriod < (datetime.now() - self._latestAddonCache.cachedAt)
+			or cacheHash and (self._latestAddonCache.cacheHash != cacheHash)

```

Comment:
same here re: cacheHash being None

```suggestion
			or cacheHash is None
			or self._compatibleAddonCache.cacheHash != cacheHash
```